pkg/kubelet/cpumanager/cpu_manager.go

```diff
@@ -273,3 +304,61 @@ func (m *manager) Admit(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitR
 		Message: "Node is experiencing CPU Pressure",
 	}
 }
+
+const (
+	reason  = "Evicted"
+	message = "The node no longer has CPUs available to run the pod"
+)
+
+func isNoCPUPod(pod *v1.Pod) bool {
+	for _, container := range pod.Spec.Containers {
+		if isNoCPUContainer(&container) {
+			return true
+		}
+	}
+	return false
+}
+
+func isNoCPUContainer(container *v1.Container) bool {
+	for resourceName, request := range container.Resources.Requests {
```

Comment:
Since container.Resources.Requests is a map, want to do a lookup here instead of iterating the entries?